pkg/controller/servicebinding/secret_test.go

```diff
@@ -38,17 +37,96 @@ func TestSecretNew(t *testing.T) {
 		f.FakeDynClient(),
 		ns,
 		name,
+		data,
+		secretOwnerReference,
 	)
 
-	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName, err := HashedResourceNameForData(name, data)
 		assert.NoError(t, err)
+		assert.Contains(t, hashedName, name)
+	})
+
+	t.Run("create", func(t *testing.T) {
```

Comment:
Another clue that `createOrUpdate` wouldn't need to be modified other than including the resource name.

Reply:
Done